src/CLI/OutputGenerator.cs

```diff
@@ -0,0 +1,152 @@
+            {
+                string version = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).ProductVersion;
+                _writer.WriteLine("Axe.Windows Accessibility Scanner (version {0})", version);
+
+                bool haveProcessName = options.ProcessName != IProcessHelper.InvalidProcessName;
+                bool haveProcessId = options.ProcessId != IProcessHelper.InvalidProcessId; 
+
+                if (haveProcessName || haveProcessId)
+                {
+                    _writer.Write("Scan Target:");
+
+                    if (haveProcessName)
+                    {
+                        _writer.Write(" Process Name = {0}", options.ProcessName);
+                    }
+                    if (haveProcessName && haveProcessId)
+                    {
+                        _writer.Write(",");
+                    }
+                    if (haveProcessId)
+                    {
+                        _writer.Write(" Process ID = {0}", options.ProcessId);
+                    }
+                    _writer.WriteLine();
+                }
+                if (!string.IsNullOrEmpty(options.ScanId))
+                {
+                    _writer.WriteLine("Scan Id = {0}", options.ScanId);
+                }
+
+                _bannerHasBeenShown = true;
+            }
+        }
+
+        private void ShowExecutionErrors(IErrorCollector errorCollector)
+        {
+            foreach (string parameterError in errorCollector.ParameterErrors)
+            {
+                _writer.WriteLine("Parameter Error: {0}", parameterError);
+            }
+
+            foreach (Exception exception in errorCollector.Exceptions)
+            {
+                _writer.WriteLine("The following exception was caught: {0}", exception);
+            }
+        }
+
+        private void ShowScanResults(IOptions options, ScanResults scanResults)
+        {
+            if (options.VerbosityLevel == VerbosityLevel.Quiet)
+            {
+                return;
+            }
+
+            if (scanResults.ErrorCount == 1)
+            {
+                _writer.WriteLine("1 error was found");
+            }
+            else
+            {
+                _writer.WriteLine("{0} errors were found", scanResults.ErrorCount);
+            }
+
+            if (options.VerbosityLevel >= VerbosityLevel.Verbose)
+            {
+                ShowVerboseResults(scanResults);
+            }
+
+            if (!string.IsNullOrEmpty(scanResults.OutputFile.A11yTest))
+            {
+                _writer.WriteLine("Results were written to \"{0}\"", scanResults.OutputFile.A11yTest);
+            }
+        }
+
+        private void ShowVerboseResults(ScanResults scanResults)
+        {
+            int errorCount = 0;
+            foreach (ScanResult scanResult in scanResults.Errors)
+            {
+                _writer.WriteLine("Error {0}: {1}", ++errorCount, scanResult.Rule.Description);
```

Comment:
Does writing an error index provide value?

Reply:
I think it does if you're trying to refer to a specific issue in the output. It's basically free to include, and I can't think of any potential problem that it would represent

Reply:
Maybe eventually we could have a parameter for it, but I don't think we need to do that now.